resources/leaflet/jquery.leaflet.js

```diff
@@ -336,7 +341,11 @@
 			this.bindClickTarget();
 
 			var layers = {};
-			$.each(options.layers.reverse(), function(index, layerName) {
+			var optionLayers = options.layers;
+			if ( this.isUserUsesDarkMode() ) {
+				optionLayers = layersDark;
+			}
+			$.each( optionLayers.reverse(), function(index, layerName) {
```

Comment:
You can avoid the state and mutation you are doing here by introducing a new function:

```js
$.each( getLayers().reverse(), function(index, layerName) {}

function getLayers() {
    if ( this.isUserUsesDarkMode() ) {
        return mw.config.get('egMapsLeafletLayersDark');
    }

   return options.layers;
}
```